src/client/Client.cc

```diff
@@ -11060,7 +11029,7 @@ void Client::ll_register_callbacks(struct ceph_client_callback_args *args)
 {
   if (!args)
     return;
-  std::scoped_lock l(client_lock);
+
```

Comment:
Build a check here so it fails/aborts (API abuse) if Client is mounted?

Reply:
Sure. We should make sure the Client is not in any state of mounted/mounting/unmounting.